Lib/socket.py

```diff
@@ -728,6 +728,78 @@ def create_connection(address, timeout=_GLOBAL_DEFAULT_TIMEOUT,
     else:
         raise error("getaddrinfo returns an empty list")
 
+def bind_socket(address, family=AF_UNSPEC, type=SOCK_STREAM, *, backlog=128,
+                reuse_port=False, flags=getattr(_socket, "AI_PASSIVE", 0)):
+    """Convenience function which creates a socket bound to *address*
+    (a 2-tuple (host, port)) and return the socket object.
+
+    If *host* is an empty string or None all network interfaces are assumed.
+
+    If *family* is AF_UNSPEC or None the address family will be
```

Comment:
I think better to don't support `None` for `family`, `AF_UNSPEC` should be enough.
In most cases user will not pass the argument for default behavior.
You don't support `None` for `type` for example.

Reply:
BTW, `AF_UNSPEC` == `0`, pretty reasonable default value.